### PR TITLE
network plugin: Add a new metadata "network:ip_address"

### PR DESCRIPTION
**Pull request by ashie**
ChangeLog: network plugin: Add a new metadata "network:ip_address"

Although the telemetry data already have the "host" identifier, it
can be determined by clients. This additional metadata will help you
to detect spoofing at server side.

It will also help you to filter the data by sender's address.

### CODE REVIEW

**Comment by ashie** [comment]
https://github.com/collectd/collectd/pull/3189#issuecomment-505281603 is addressed.

File: src/network.c [hunk 1]
```diff
@@ -438,6 +440,27 @@ static int network_dispatch_values(value_list_t *vl, /* {{{ */
     }
   }
 
+  if (address != NULL) {
+    char host[48];
+    status = getnameinfo((struct sockaddr *)address,
+                         sizeof(struct sockaddr_storage), host, sizeof(host),
+                         NULL, 0, NI_NUMERICHOST | NI_NUMERICSERV);
+    if (status != 0) {
+      ERROR("network plugin: meta_data_add_string failed.");
```
Comment (mrunge) [inline src/network.c]: Please change the error message here. Otherwise, it is the same as in line 457,

**Comment by dago** [comment]
@ashie Would you mind adjusting the error messages to be different? This is useful when associating log messages with code.

**Comment by ashie** [comment]
Oops, sorry I missed the comment.
I'll address it.

**Comment by ashie** [comment]
I've addressed it.

**Comment by dago** [comment]
@ashie There is now a failure reported from CI that `network_test.c` lacks arguments as you also changed the signature of `parse_packet`. Would you mind adjusting this also and making sure that the testsuite still runs smooth?

**Comment by ashie** [comment]
> @ashie There is now a failure reported from CI that `network_test.c` lacks arguments as you also changed the signature of `parse_packet`. Would you mind adjusting this also and making sure that the testsuite still runs smooth?

Thank you for your quick reply, I've also noticed it just now. I'll address it.
The cause is that the patch is based on old code.
Can I rebase this PR?

**Comment by dago** [comment]
Sure, just rebase and force-push the branch.

**Comment by ashie** [comment]
> Sure, just rebase and force-push the branch.

Done. I confirmed that `make distcheck` passed on top of the current master.

**Comment by ashie** [comment]
Thanks!

**Pull request merged**